frontend/src/lib/utils/product-intents.ts

```diff
@@ -0,0 +1,57 @@
+import api from 'lib/api'
+
+import type { ProductKey, TeamType } from '~/types'
+
+export enum ProductIntentContext {
+    // Cross Sell
+    CROSS_SELL = 'cross sell',
+
+    // Onboarding
+    ONBOARDING_PRODUCT_SELECTED_PRIMARY = 'onboarding product selected - primary',
+    ONBOARDING_PRODUCT_SELECTED_SECONDARY = 'onboarding product selected - secondary',
+
+    // Data Warehouse
+    SELECTED_CONNECTOR = 'selected connector',
+
+    // Experiments
+    EXPERIMENT_CREATED = 'experiment created',
+
+    // Feature Flags
+    FEATURE_FLAG_CREATED = 'feature flag created',
+}
+
+export type ProductIntentProperties = {
+    product_type: ProductKey
+    intent_context: ProductIntentContext
+    metadata?: Record<string, unknown>
+}
+
+export function addProductIntent(properties: ProductIntentProperties): Promise<TeamType> {
+    return api.productIntents.update(properties)
+}
+
+export type ProductCrossSellMetadata = Record<string, unknown>
+
+export enum ProductCrossSellLocation {
+    TAXONOMIC_FILTER_EMPTY_STATE = 'taxonomic_filter_empty_state',
+}
+
+export type ProductCrossSellProperties = {
+    from: ProductKey
+    to: ProductKey
+    location: ProductCrossSellLocation
+    metadata?: ProductCrossSellMetadata
+}
+
+export function addProductCrossSell(properties: ProductCrossSellProperties): Promise<TeamType> {
```

Comment:
I think there might be opportunity to be confused here about when to use which type. Do I use a `crossSell` or a regular `intent`? Do I get an intent with a crossSell?

So maybe...

```suggestion
export function addProductIntentViaCrossSell(properties: ProductCrossSellProperties): Promise<TeamType> {
```

Reply:
Yeah this is nicer, will do `addProductIntentForCrossSell`